File: 0435-non-overlapping-intervals/0435-non-overlapping-intervals.py
class Solution:
    def eraseOverlapIntervals(self, intervals: List[List[int]]) -> int:
        
        intervals.sort(key=lambda x:x[1])
        prev = intervals[0][1]
        count = 0

        for i in range(1, len(intervals)):
            if prev > intervals[i][0]:
                count += 1
            else:
                prev = intervals[i][1]

        return count

File: 0435-non-overlapping-intervals/0435-non-overlapping-intervals.java
class Solution {
    public int eraseOverlapIntervals(int[][] intervals) {

        int count = 0;
        Arrays.sort(intervals, (a, b) -> (a[1] - b[1]));

        int prev = intervals[0][1];
        for(int i = 1; i < intervals.length; i++){
            if(intervals[i][0] < prev){
                count++;
            } else {
                prev = intervals[i][1];
            }
        }

        return count;
    }
}